src/pools/xaveFxPool/fxPool.ts

```diff
@@ -0,0 +1,276 @@
+        this.beta = parseFixed(beta, 18);
+        this.lambda = parseFixed(lambda, 18);
+        this.delta = parseFixed(delta, 18);
+        this.epsilon = parseFixed(epsilon, 18);
+    }
+    updateTotalShares: (newTotalShares: BigNumber) => void;
+    mainIndex?: number | undefined;
+    isLBP?: boolean | undefined;
+
+    _calcTokensOutGivenExactBptIn(bptAmountIn: BigNumber): BigNumber[] {
+        // Will copy over other implementations, not supporting BPT tokens atm
+        return new Array(this.tokens.length).fill(Zero);
+    }
+
+    _calcBptOutGivenExactTokensIn(amountsIn: BigNumber[]): BigNumber {
+        // Will copy over other implementations, not supporting BPT tokens atm
+        return Zero;
+    }
+
+    parsePoolPairData(tokenIn: string, tokenOut: string): FxPoolPairData {
+        const tokenIndexIn = this.tokens.findIndex(
+            (t) => getAddress(t.address) === getAddress(tokenIn)
+        );
+        if (tokenIndexIn < 0) throw 'Pool does not contain tokenIn';
+        const tI = this.tokens[tokenIndexIn];
+        const balanceIn = tI.balance;
+        const decimalsIn = tI.decimals;
+
+        const tokenIndexOut = this.tokens.findIndex(
+            (t) => getAddress(t.address) === getAddress(tokenOut)
+        );
+
+        if (tokenIndexOut < 0) throw 'Pool does not contain tokenOut';
+        const tO = this.tokens[tokenIndexOut];
+        const balanceOut = tO.balance;
+        const decimalsOut = tO.decimals;
+
+        const poolPairData: FxPoolPairData = {
+            id: this.id,
+            address: this.address,
+            poolType: this.poolType,
+            tokenIn: tokenIn,
+            tokenOut: tokenOut,
+            decimalsIn: Number(decimalsIn),
+            decimalsOut: Number(decimalsOut),
+            balanceIn: parseFixed(balanceIn, decimalsIn),
+            balanceOut: parseFixed(balanceOut, decimalsOut),
+            swapFee: this.swapFee,
+            alpha: this.alpha,
+            beta: this.beta,
+            lambda: this.lambda,
+            delta: this.delta,
+            epsilon: this.epsilon,
+            tokenInRate: parseFixed(this.tokens[tokenIndexIn].latestFXPrice!),
+            tokenOutRate: parseFixed(this.tokens[tokenIndexOut].latestFXPrice!),
+        };
+
+        return poolPairData;
+    }
+
+    // Normalized liquidity is an abstract term that can be thought of the
+    // inverse of the slippage. It is proportional to the token balances in the
+    // pool but also depends on the shape of the invariant curve.
+    // As a standard, we define normalized liquidity in tokenOut
+    getNormalizedLiquidity(poolPairData: FxPoolPairData): OldBigNumber {
+        return universalNormalizedLiquidity(
+            this._derivativeSpotPriceAfterSwapExactTokenInForTokenOut(
+                poolPairData,
+                ZERO
+            )
+        );
+    }
+
+    // There will be no limit amount to swap.
+    getLimitAmountSwap(
+        poolPairData: FxPoolPairData,
```

Comment:
There's a bunch of warning being shown for mismatch in poolPairData, e.g.:
```
Property 'getLimitAmountSwap' in type 'FxPool' is not assignable to the same property in base type 'PoolBase<PoolPairBase>'.
  Type '(poolPairData: FxPoolPairData, swapType: SwapTypes) => BigNumber' is not assignable to type '(poolPairData: PoolPairBase, swapType: SwapTypes) => BigNumber'.
    Types of parameters 'poolPairData' and 'poolPairData' are incompatible.
      Type 'PoolPairBase' is not assignable to type 'FxPoolPairData'.
```
Could you please update to fix those?

Reply:
does this still exist on your end? i made some changes and it does appear already